Spring quickstart

slaskawi opened the pull request:
Hi!

Please take a look at the implementation of Spring integration Quickstart. I'll backport it to the production branch as soon as upstream gets merged.

Implementation details:
* Used Spring boot to simplify Spring setup.
* Created Ember JS frontend which communicates with backend via REST interfaces (based on Spring MVC).
* Used Patternfly look and feel to be consistent with Red Hat brand.

Thanks!
Sebastian

slaskawi commented:
Hey @mgencur @tristantarrant !

CI passes with this PR, could you please take a look at the content? If everything is fine - I would like to push it and then prepare a backport for 6.4.

Thanks!

tristantarrant commented on spring/README.md:
```diff
@@ -0,0 +1,51 @@
+
+Spring-jdg is an example that shows how to use Infinispan or JBoss Data Grid together with Spring.
+
+In order to simplify the Spring Bootstrapping this project uses Spring Boot.
```
s/Bootstrapping/bootstrapping/

slaskawi replied:
done

tristantarrant commented on spring/README.md:
```diff
@@ -0,0 +1,51 @@
+
+        mvn clean package spring-boot:run
+
+2. Navigate to <http://localhost:8080> and follow instructions in about page.
```
follow the instructions

slaskawi replied:
done

tristantarrant commented on spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/bootstrap/ServletInitializer.java:
```diff
@@ -0,0 +1,18 @@
+ * @author Sebastian Laskawiec
+ */
+public class ServletInitializer extends SpringBootServletInitializer {
+
```
Weird indentation

slaskawi replied:
converted tabs to spaces

tristantarrant commented on spring/src/main/java/org/jboss/as/quickstarts/datagrid/spring/core/client/ClientCache.java:
```diff
@@ -0,0 +1,40 @@
+public class ClientCache {
+
+    private Cache cache;
+    private AdvancedCache<Object, Collection<Client>> nativeCache;
```
Why is the key an Object, can't we qualify it better ?

slaskawi replied:
Switched to `AdvancedCache<String, Collection<Client>>`

slaskawi commented:
Comments addressed. 

I also changed the following things:
* removed JFairy dependency (it would bite us later on)
* removed Actuator and WebSockets from the classpath (Wildfly deployment speedup + removed warnings)
* updated readme information and suggested running it on WildFly or EAP
* Updated Quickstart version in `pom.xml`

slaskawi commented:
@mgencur, @tristantarrant Could you please take a look at this?

slaskawi commented:
Integrating

slaskawi commented:
Integrated
